frontend/src/main/scala/bloop/data/Project.scala

```diff
@@ -271,18 +273,29 @@ object Project {
       (baseOptions ++ baseSemanticdbOptions ++ pluginOption).distinct
     }
 
+    def enableDottySemanticDB(options: List[String]) = {
+      val ysemanticdb = if (!options.contains("-Ysemanticdb")) List("-Ysemanticdb") else Nil
+      val sourceRoot =
+        if (!options.contains("-sourceroot")) List("-sourceroot", workspaceDir.toString()) else Nil
+      ysemanticdb ++ sourceRoot
+    }
+
     def enableRangePositions(options: List[String]): List[String] = {
       val hasYrangepos = options.exists(_.contains("-Yrangepos"))
-      val isDotty = project.scalaInstance.exists(_.isDotty)
       if (hasYrangepos || isDotty) options else options :+ "-Yrangepos"
     }
 
     val projectWithRangePositions =
       project.copy(scalacOptions = enableRangePositions(project.scalacOptions))
+
+    val options = projectWithRangePositions.scalacOptions
+
     semanticDBPlugin match {
+      case None if isDotty =>
```

Comment:
nit: instead of guarding in the pattern match, I find it more readable to put the guard inside the body
```suggestion
      case None =>
        if (isDotty) ...
        else ...
```